### PR TITLE
feat(python): release GIL for common functions

### PR DESCRIPTION
**Pull request by franz101**
# Description
Inspired by the previous request to release the GIL added to most common functions.
# Related Issue(s)
https://github.com/delta-io/delta-rs/issues/2234

If some function are not supported / should not be released or I forgot to add the function, happy to modify the PR

### CODE REVIEW

**Comment by franz101** [comment]
ah something messed up my formatter (`make format`)

**Comment by mrocklin** [comment]
I'm excited for this.  Thanks for working on it @franz101 !  Looking briefly it looks like the linter is sad.

File: python/src/lib.rs [hunk 1]
```diff
@@ -282,75 +282,81 @@ impl RawDeltaTable {
...
 
-        let (table, metrics) = rt()?
-            .block_on(cmd.into_future())
-            .map_err(PythonError::from)?;
-        self._table.state = table.state;
-        Ok(metrics.files_deleted)
+            let (table, metrics) = rt()?
+                .block_on(cmd.into_future())
+                .map_err(PythonError::from)?;
+            self._table.state = table.state;
```
Comment (wjones127) [inline python/src/lib.rs:311]: Could we move updating the table state outside of the `allow_threads` call? this requires mutating `self`, which should be done while holding the GIL.

Same thing applies to other methods.

**Comment by adriangb** [comment]
For the cases when you need the GIL, could you also release it and re-acquire it when needed, if the fraction of time it's not needed is large?

**Pull request closed without merging**